index.js

```diff
@@ -73,13 +74,25 @@ exports.register = (server, options, next) => {
       if (!continueProcessing) {
         return;
       }
-      updateHooks(server, settings, collection, (err) => {
-        if (err) {
-          server.log(['hapi-hooks', 'error'], err);
+      logHooks(server, settings, collection, (countErr, current) => {
+        if (countErr) {
+          server.log(['hapi-hooks', 'error'], countErr);
         }
-        if (continueProcessing) {
```

Comment:
The only thing I was saying is that you could move this outside of the call back and then it would be non blocking.